### PR TITLE
adds checks to edit BLIs

### PR DESCRIPTION
**Pull request by fpigeonjr**
## What changed
- allows BLIs that are  in DRAFT and IN_REVIEW and PLANNED to be edited from Agreement detail BLI table
- updates prop names to more generic

Does Not Include:
- Backend changes to reflect BLI status back to DRAFT
- e2e tests
- checks to see if you are authorized

## Issue

#1374 

## How to test

- goto Agreement 2 // has BLIs > IN_Review
- add a DRAFT BLI and send to review
- try to edit that BLI // it should allow you to
- save the changes and should see success Alert
- TODO: Backend changes coming to reflect BLI status back to DRAFT

### CODE REVIEW

**Comment by jonnalley** [comment]
@fpigeonjr so what about planned status?  is that already in place or will that be done in a different PR ?

**Comment by fpigeonjr** [comment]
> @fpigeonjr so what about planned status? is that already in place or will that be done in a different PR ?

Did not account for PLANNED BLIs but will take of this use case with this PR

**Comment by jonnalley** [comment]
> Did not account for PLANNED BLIs but will take of this use case with this PR

Gotcha thanks @fpigeonjr. The nature of the story is more about planned BLs than anything else.

**Comment by fpigeonjr** [comment]
I guess I assume that DRAFT, IN_REVIEW, and PLANNED BLIs will get the same treatment

**Comment by jonnalley** [comment]
> I guess I am assuming that DRAFT, IN_REVIEW, and PLANNED BLIs will get the same treatment

yup, except if a BL is already in the latter two, it would then move "down" to draft after the change.

File: frontend/src/components/UI/PreviewTable/PreviewTable.jsx [hunk 1]
```diff
@@ -53,6 +53,10 @@ export const PreviewTable = ({
         }
         let feeTotal = bl?.amount * bl?.psc_fee_amount;
         let total = bl?.amount + feeTotal;
+        const isBudgetLineDraft = bl?.status === "DRAFT";
+        const isBudgetLineInReview = bl?.status === "UNDER_REVIEW";
+        const isBudgetLinePlanned = bl?.status === "PLANNED";
```
Comment (fpigeonjr) [inline frontend/src/components/UI/PreviewTable/PreviewTable.jsx:58]: @jonnalley PLANNED is now accounted for

**Review by johndeange — APPROVED** [review]

**Review by stevtek — APPROVED** [review]

**Pull request merged**